### PR TITLE
Inspect order issue

### PR DESCRIPTION
**Pull request by gvszg**

### CODE REVIEW

File: app/models/concerns/info_inspector.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  def inspec_is_blacklist
+    if is_in_blacklist? || is_invalid_data_format?
+      self.update_attribute(:is_blacklist, true)
+    end
+  end
+
+  private
+
+  def is_in_blacklist?
```
Comment (varbark) [inline app/models/concerns/info_inspector.rb]: def in_blacklist? 即可

File: app/models/concerns/info_inspector.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    OrderBlacklist.email_blacklists.include?(self.ship_email) || OrderBlacklist.phone_blacklists.include?(self.ship_phone)
+  end
+
+  def is_invalid_data_format?
```
Comment (varbark) [inline app/models/concerns/info_inspector.rb]: 同上

File: db/migrate/20160617031753_add_is_blacklist_to_order_infos.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class AddIsBlacklistToOrderInfos < ActiveRecord::Migration
+  def change
+    add_column :order_infos, :is_blacklist, :boolean
```
Comment (varbark) [inline db/migrate/20160617031753_add_is_blacklist_to_order_infos.rb]: is_blacklist   =>  blacklist?

File: app/models/order_blacklist.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class OrderBlacklist < ActiveRecord::Base
+  OrderBlacklist::Email_Format = /\A[a-z0-9._%+]+@[a-z0-9.]+\.[a-z]{2,4}\Z/i
```
Comment (StevenKo) [inline app/models/order_blacklist.rb]: check email regular expression

File: app/models/order_blacklist.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+  OrderBlacklist::Phone_Format = /\A(0)(9)+\d{8}\Z/
```
Comment (StevenKo) [inline app/models/order_blacklist.rb]: (04)7585506
7585506
+886-2-1234-5678
???

File: app/models/concerns/order_concern.rb [hunk 6]
```diff
@@ -12,6 +13,13 @@ def notify_user_if_arrive_store
     end
   end
 
+  def create_order_blacklist
+    if (status_changed? && status == "未取訂貨")
+      create_new_blacklist
```
Comment (StevenKo) [inline app/models/concerns/order_concern.rb]: create_order_blacklist , create_new_blacklist ??? confusing

File: app/models/concerns/order_concern.rb [hunk 7]
```diff
@@ -12,6 +13,13 @@ def notify_user_if_arrive_store
     end
   end
 
+  def create_order_blacklist
+    if (status_changed? && status == "未取訂貨")
```
Comment (StevenKo) [inline app/models/concerns/order_concern.rb]: do need status_changed?

File: app/models/concerns/info_inspector.rb [hunk 8]
```diff
@@ -0,0 +1,23 @@
+module InfoInspector
+  extend ActiveSupport::Concern
+
+  included do
+    after_create :inspec_is_blacklist
```
Comment (StevenKo) [inline app/models/concerns/info_inspector.rb]: after_update or after_create??

File: db/schema.rb [hunk 9]
```diff
@@ -278,6 +286,7 @@
     t.integer  "ship_store_id",   limit: 4
     t.string   "ship_store_name", limit: 255
     t.string   "ship_email",      limit: 255
+    t.boolean  "in_blacklist"
```
Comment (StevenKo) [inline db/schema.rb]: is_blacklisted

File: app/models/order.rb [hunk 10]
```diff
@@ -31,6 +31,10 @@ def self.search_by_phone_and_email(phone, email)
     self.joins(:info).where(order_infos: {ship_phone: phone, ship_email: email}).recent
   end
 
+  def blacklist?
```
Comment (StevenKo) [inline app/models/order.rb]: delegate?

Comment (StevenKo) [inline app/models/order.rb]: delegate?

File: app/models/concerns/info_inspector.rb [hunk 11]
```diff
@@ -0,0 +1,23 @@
+module InfoInspector
+  extend ActiveSupport::Concern
+
+  included do
+    after_save :inspec_info_if_blacklisted
```
Comment (StevenKo) [inline app/models/concerns/info_inspector.rb]: set_blacklisted_if_not_well_formed_or_in_blacklist

File: app/models/concerns/order_concern.rb [hunk 12]
```diff
@@ -1,8 +1,9 @@
 module OrderConcern
   extend ActiveSupport::Concern
 
-  included do    
+  included do
     after_update :notify_user_if_arrive_store
+    after_update :check_order_if_blacklisted
```
Comment (StevenKo) [inline app/models/concerns/order_concern.rb]: check_order_if_blacklisted ......
set_blacklisted_if_not_pickup

File: app/models/order_blacklist.rb [hunk 13]
```diff
@@ -0,0 +1,7 @@
+class OrderBlacklist < ActiveRecord::Base
+  OrderBlacklist::Email_Format = /\A([\w+\-].?)+@[a-z\d\-]+(\.[a-z]+)*\.[a-z]+\z/i
+  OrderBlacklist::Phone_Format = /\A(\+\d{3}-)?(\(?(0)?([2-9])\)?)(-)?(\d{2})?(-)?\d{3,4}(-)?\d{3,4}\z/
+
```
Comment (StevenKo) [inline app/models/order_blacklist.rb:6]: 列出可以和不可以的電話號碼

File: app/models/concerns/info_inspector.rb (hunk 11 above)
Comment (StevenKo) [inline app/models/concerns/info_inspector.rb]: set_blacklisted_by_checking_info

**Pull request merged**